File: app/src/main/java/com/qos/testnet/ui/dashboard/IndividualTestFragment.java
package com.qos.testnet.ui.dashboard;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.qos.myapplication.databinding.FragmentIndividualTestBinding;

public class IndividualTestFragment extends Fragment {

    private FragmentIndividualTestBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        IndividualTestViewModel individualTestViewModel =
                new ViewModelProvider(this).get(IndividualTestViewModel.class);

        binding = FragmentIndividualTestBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        final TextView textView = binding.textDashboard;
        individualTestViewModel.getText().observe(getViewLifecycleOwner(), textView::setText);
        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/qos/testnet/ui/dashboard/IndividualTestFragment.kt
package com.qos.testnet.ui.dashboard

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.qos.testnet.databinding.FragmentIndividualTestBinding
import com.qos.testnet.utils.AdapterResults
import kotlinx.coroutines.launch

class IndividualTestFragment : Fragment() {

    private var _binding: FragmentIndividualTestBinding? = null
    private val binding get() = _binding!!

    private lateinit var adapter: AdapterResults

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        _binding = FragmentIndividualTestBinding.inflate(inflater, container, false)
        val root: View = binding.root

        val factory = IndividualTestViewModelFactory(requireContext())
        val individualTestViewModel =
            ViewModelProvider(this, factory)[IndividualTestViewModel::class.java]

        lifecycleScope.launch {
            individualTestViewModel.fetchData()
        }

        adapter = AdapterResults()
        binding.recyclerView.adapter = adapter
        binding.recyclerView.layoutManager = LinearLayoutManager(context)

        individualTestViewModel.fetchedData.observe(viewLifecycleOwner) { dataList ->
            adapter.submitList(dataList)
        }

        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}